### PR TITLE
Athena - Get temporary credentials from the conn_id

### PR DESCRIPTION
**Pull request by octiva**
## Description

Passes the `conn_id` to the `AwsGenericHook` and uses `get_credentials()`, which handles the creation of a session, credentials, freezing of credentials & also masking. 

[See get_credentials() docs here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/base_aws/index.html#airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook.get_credentials)

## Related Issue(s)

#691 

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by pixie79** [comment]
Do you mean dbt-Athena or dbt-athena-community. Dbt-Athena is the original module but the community version is much improved and optimised. https://github.com/dbt-athena/dbt-athena

File: dev/dags/example_athena_profile.py
Comment (jbandoro) [inline dev/dags/example_athena_profile.py:1]: Cosmos doesn't have example DAGs for all of the profile mappings it supports, so I would suggest removing this file. The DAGs in `dev/dags` are run for integration testing and this one I think would fail.

Docs for all of the profile mappings are [here](https://astronomer.github.io/astronomer-cosmos/profiles/index.html) which are sourced from the profile mapping classes. I'd suggest updating the sphinx docstring of the the `AthenaAccessKeyProfileMapping` class with any useful details that would be helpful for end users in using the profile.

**Review by jbandoro — COMMENTED** [review]
Thanks @octiva for the contribution!

I'm not familiar with Athena access keys, so if anyone from #691 wants to chime in here it would be helpful. My main comment below is if we should keep the current profile mapping of the Airflow connection, are there cases where users would still want to use the Airflow connection params, or is it broken and this PR fixes it?

File: dev/dags/example_athena_profile.py
Comment (octiva) [reply dev/dags/example_athena_profile.py:1]: Thanks that is a great point. I will remove, and also update the docs.

**Comment by octiva** [comment]
@jbandoro Thanks for the feedback. If the user has an Airflow connection with no `extra.role_arn` provided, their credentials  (secret + key id) used in the DBT profile will be unchanged, except there will now always be a `aws_session_token`.

File: cosmos/profiles/athena/access_key.py
Comment (octiva) [reply cosmos/profiles/athena/access_key.py:60]: Good call out, we can safely remove those. 

The AwsGenericHook handles the secret masking, so there is no need to store them in the envs, we can also remove that logic.  [AWS docs](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/base_aws/index.html#airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook.get_credentials)

I also want to wait for @benjamin-awd's feedback before remove too much of the existing logic.

Comment (octiva) [reply cosmos/profiles/athena/access_key.py]: @jbandoro removed `aws_secret_access_key`, `aws_session_token` & `aws_access_key_id` from `airflow_param_mapping` as these always come from the hook. 

I dont think we can remove them from the `required_field`, as they are a check on the connection that we will be able to create a profile, mainly when creating via the automatic_mapping function. We could use the `test_connection()` to ensure the credentials are usable, however that doesnt really fit the current pattern for profile from Airflow connections.

File: dev/dags/example_athena_profile.py
Comment (octiva) [reply dev/dags/example_athena_profile.py:1]: @jbandoro Removed all updates to `/dev`. Also updated the docs in the sphinx docstring to update what is happening and where we can find information.

File: cosmos/profiles/athena/access_key.py
Comment (jbandoro) [reply cosmos/profiles/athena/access_key.py]: Thanks!

You're right about the `required_fields` that they should be kept. Though they might need to be removed from the `secret_fields` or else trying to claim the connection will fail:

https://github.com/astronomer/astronomer-cosmos/blob/7b3e8ab93eda1dfd233ad201a231fb5033615bd7/cosmos/profiles/base.py#L90-L98

The unit tests should confirm this though once the GH actions are fixed.

File: cosmos/profiles/athena/access_key.py
Comment (benjamin-awd) [inline cosmos/profiles/athena/access_key.py:76]: Should the AWS Airflow provider (`apache-airflow-providers-amazon`) be listed somewhere as an optional dependency in the pyproject.toml?

**Comment by benjamin-awd** [comment]
LGTM, I tried this out today and it seems to work for my use-case -- unfortunately I don't have time to look into the memory issue and verify that this works at scale, but that's hopefully going to be resolved with the use of the AWS hook. Ran a couple of models concurrently and didn't run into any issues.

File: cosmos/profiles/athena/access_key.py
Comment (octiva) [reply cosmos/profiles/athena/access_key.py:76]: Thanks @benjamin-awd, ive added aws as a optional-dependency in the pyproject.toml

File: pyproject.toml
Comment (tatiana) [inline pyproject.toml:124]: Since, ATM, this dependency is specific to Athena, I suggest we add it to the previously existing:
```
dbt-athena = [
```

File: tests/profiles/athena/test_athena_access_key.py
Comment (tatiana) [inline tests/profiles/athena/test_athena_access_key.py:201]: Before, we were obfuscating the secrets and setting them in environment variables. This behaviour should continue for temporary credentials as well.

File: tests/profiles/athena/test_athena_access_key.py
Comment (tatiana) [inline tests/profiles/athena/test_athena_access_key.py:188]: Before, we were obfuscating the secrets and setting them in environment variables. This behaviour should continue for temporary credentials as well.

File: tests/profiles/athena/test_athena_access_key.py
Comment (tatiana) [inline tests/profiles/athena/test_athena_access_key.py:162]: Before, we were obfuscating the secrets and setting them in environment variables. This behaviour should continue for temporary credentials as well.

**Review by tatiana — COMMENTED** [review]
Thanks for your contribution, @octiva! It is an excellent improvement to leverage the hook method instead of directly retrieving the values from the environment variable.

Before, we were obfuscating the secrets and setting them as environment variables. I suggest we keep this behaviour, even if the credentials are temporary. The description of your PR states that the Airflow `get_credentials` method masks those secrets. From what I understood from the docs:
>  By use this method also secret_key and token will mask in tasks logs.
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/base_aws/index.html#airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook.get_credentials

In other words, it seems they are masked only in the logs - not in the `profiles.yml` file that Cosmos creates.

Assuming the value we wrote in the `profiles.yml` file was masked, the dbt commands using them would fail to run - since they would try to use invalid credentials.

File: tests/profiles/athena/test_athena_access_key.py
Comment (octiva) [reply tests/profiles/athena/test_athena_access_key.py]: Great call out, sorry I missed it. Ive fixed it up now. Let me know what you think of the solution.

**Comment by octiva** [comment]
@tatiana I am unable to write a test for the `_get_temporary_credentials`method, as I am unable to import `airflow.providers.amazon` via the test dependencies as we get some dependency conflicts when using Python 3.8/3.9 + Airflow 2.3. 

The amazon provider at 8.0.0 is compatible with 2.3 and python down to 3.7.

https://pypi.org/project/apache-airflow-providers-amazon/8.0.0/

I tried many ways to fix this on the weekend, but could not. I ended up resolving this by removing the dependency and mocking the class function. 

This has the unfortunately side-affect of reducing our test coverage

**Comment by jbandoro** [comment]
Thanks @octiva addressing all the feedback! For your question below on test coverage:

> I tried many ways to fix this on the weekend, but could not. I ended up resolving this by removing the dependency and mocking the class function.
> 
> This has the unfortunately side-affect of reducing our test coverage

You can get test coverage for the `_get_temporary_credential` method by instead patching the provider module that can't be imported, like in the examples [here](https://github.com/astronomer/astronomer-cosmos/blob/9fcee8e785ea3bcb0d19e8d36f3f5b94bedafc98/tests/operators/test_local.py#L439) and [here](https://github.com/astronomer/astronomer-cosmos/blob/9fcee8e785ea3bcb0d19e8d36f3f5b94bedafc98/tests/test_export.py#L76-L83).

It might be easier to do this patching in a fixture and reuse it in all of tests that require it.

**Comment by spencer-horton-wcq** [comment]
@jbandoro Thanks for the hint. I had attempted this, but was doing it at the wrong level, and now ive got it working quite nicely. Let me know what you think

**Comment by octiva** [comment]
reposting from correct account @jbandoro Thanks for the hint. I had attempted this, but was doing it at the wrong level, and now ive got it working quite nicely. Let me know what you think 🚀

File: tests/profiles/athena/test_athena_access_key.py
Comment (jbandoro) [inline tests/profiles/athena/test_athena_access_key.py:71]: Do you think we need both the `mock_athena_assume_role_conn` and the `mock_athena_conn` fixture herd, could we keep only 1. 

It looks like the only difference is that one has `"role_arn"` field in its extra which isn't tested or used in the tests.

**Review by jbandoro — APPROVED** [review]
Thanks @octiva! Mocking the hook looks much better. Approving with some minor comments below to address before merging.

File: tests/profiles/athena/test_athena_access_key.py
Comment (spencer-horton-wcq) [reply tests/profiles/athena/test_athena_access_key.py:45]: Agreed, removed

File: cosmos/profiles/athena/access_key.py
Comment (spencer-horton-wcq) [reply cosmos/profiles/athena/access_key.py]: thanks !

Comment (octiva) [reply cosmos/profiles/athena/access_key.py]: thanks !

File: tests/profiles/athena/test_athena_access_key.py
Comment (octiva) [reply tests/profiles/athena/test_athena_access_key.py:45]: Agreed, removed

**Pull request merged**
